code/mob/living/carbon/human/monkeys.dm

```diff
@@ -258,6 +260,15 @@
 			return
 		..()
 		if (src.ai_state == 0)
+			if (src.equipped() && istype(src.equipped(),/obj/item/implant/projectile/body_visible/dart/bardart))
+				for (var/obj/item/reagent_containers/balloon/balloon in view(7, src))
+					src.throw_item(balloon, list("npc_throw"))
+					src.ai_delay_move = TRUE
+					break
+			if (!src.equipped() && !src.ai_delay_move)
```

Comment:
```suggestion
			else if (!src.ai_delay_move && !src.equipped())
```
checking a bool is faster than running a proc, so it goes first. Also throw an else on there so the monkey can't throw a dart and pick up another in the same tick.